homeassistant/components/somfy/climate.py

```diff
@@ -0,0 +1,216 @@
+"""Support for Somfy Covers."""
+
+import logging
+from typing import Any, Dict, List, Optional
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_COOL,
+    HVAC_MODE_HEAT,
+    PRESET_AWAY,
+    PRESET_HOME,
+    PRESET_NONE,
+    PRESET_SLEEP,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.helpers import temperature
+from pymfy.api.devices.thermostat import (
+    DurationType,
+    RegulationState,
+    Thermostat,
+    HvacState,
+    TargetMode,
+)
+from pymfy.api.devices.category import Category
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.const import ATTR_BATTERY_LEVEL, ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from . import SomfyEntity
+from .const import API, COORDINATOR, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORTED_CATEGORIES = {Category.HVAC.value}
+
+PRESET_FROST_GUARD = "Frost Guard"
+PRESET_GEOFENCING = "Geofencing"
+PRESET_MANUAL = "Manual"
```

Comment:
About translations, what can I do?

Reply:
I am not a translation expert at all, but generally my impression is there is a way for integrations to set their own strings for translation and override existing platforms: https://developers.home-assistant.io/docs/internationalization/core

However I notice the [ClimateEntity docs](https://developers.home-assistant.io/docs/internationalization/core] say "There are a couple of built-in presets that will offer translations, but you're also allowed to add custom presets." which implies you don't get translations for your own preset modes.

I found some frontend translations here:
https://github.com/home-assistant/frontend/blob/0894a30f138ea8c9dd090124e30d6859076b3519/translations/frontend/en.json#L42

Leaving for someone else to answer... though my hunch is that there might need to be broad usage before adding more presets.

Reply:
Thank you for the frontend link. The translation seems to be done indeed on the frontend site.

With a little search, we can see some existing manual presets already exist: https://github.com/home-assistant/core/search?p=1&q=filename%3Aclimate.py+%22manual%22

Perhaps should I open a PR to discuss it?

Reply:
Yeah that sounds ok to me, though not my area of expertise, someone more experience in this code base could chime in...